userspace/falco/configuration.cpp

```diff
@@ -31,6 +31,11 @@ limitations under the License.
 #include "configuration.h"
 #include "logger.h"
 
+#include <re2/re2.h>
+
+static re2::RE2 ipv4_address_re("^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$");
```

Comment:
Can we add some tests around this? Eg here: https://github.com/falcosecurity/falco/tree/master/unit_tests/falco/app/actions

Reply:
+1 on some unit tests.

We more recently refactored a few things to make unit testing easier, however you will notice we still have a lot of work ahead, especially in terms of coverage.

Therefore, as minimum test requirement testing the regex only seems fair, for example checkout the prometheus interval test https://github.com/falcosecurity/falco/blob/1609ee85596cd4e57c15423e17a54fe1eb329092/unit_tests/engine/test_falco_utils.cpp#L41

We also have a newer repo here https://github.com/falcosecurity/testing.

As said additional tests and more future contribution in this regard would be appreciated, but only a regex test should be the requirement to get this PR merged.

Thanks in advance!

Reply:
Feel free to implement the best test. Happy to provide more feedback once it is up. Just wanted to communicate the absolute minimum test requirement beforehand as you will see that currently not all configurations are tested in this repo's unit test suite :)

Reply:
EDIT: I see it's up, one sec!